core/src/components/refresher/refresher.tsx

```diff
@@ -729,11 +731,35 @@ export class Refresher implements ComponentInterface {
         scrollStyle.transform = backgroundStyle.transform = y > 0 ? `translateY(${y}px) translateZ(0px)` : '';
         scrollStyle.transitionDuration = backgroundStyle.transitionDuration = duration;
         scrollStyle.transitionDelay = backgroundStyle.transitionDelay = delay;
-        scrollStyle.overflow = overflowVisible ? 'hidden' : '';
+        if (overflowVisible) {
+          scrollStyle.overflow = 'hidden';
+        } else {
+          this.restoreOverflowStyle();
+        }
       }
     });
   }
 
+  private memoizeOverflowStyle() {
+    if (this.scrollEl) {
+      const { overflow, overflowX, overflowY } = this.scrollEl.style;
+      this.overflowStyles = {
+        overflow: overflow ?? '',
+        overflowX: overflowX ?? '',
+        overflowY: overflowY ?? '',
+      };
+    }
+  }
+
+  private restoreOverflowStyle() {
+    if (this.overflowStyles !== undefined && this.scrollEl !== undefined) {
+      const { overflow, overflowX, overflowY } = this.overflowStyles;
+      this.scrollEl.style.overflow = overflow;
+      this.scrollEl.style.overflowX = overflowX;
+      this.scrollEl.style.overflowY = overflowY;
+    }
```

Comment:
Do we need to clear `this.overflowStyles` when done?

Reply:
I did some initial testing and clearing the overflow styles does not impact the experience, so I updated the implementation to clear the styles after assigning them.